include/tbbexec/tbb_thread_pool.hpp

```diff
@@ -0,0 +1,413 @@
+                                                                          Shape,
+                                                                          Fun,
+                                                                          Sender,
+                                                                          Receiver>) {
+                return bulk_op_state_t<Self, Receiver>{self.pool_, self.shape_, self.fun_, ((Self &&) self).sndr_,
+                                                       (Receiver &&) rcvr};
+            }
+
+            template <stdexec::__decays_to<bulk_sender> Self, class Env>
+            friend auto tag_invoke(stdexec::get_completion_signatures_t, Self&&, Env)
+                -> stdexec::dependent_completion_signatures<Env>;
+
+            template <stdexec::__decays_to<bulk_sender> Self, class Env>
+            friend auto tag_invoke(stdexec::get_completion_signatures_t, Self&&, Env)
+                -> completion_signatures<Self, Env>
+                requires true
+            ;
+
+            template <stdexec::tag_category<stdexec::forwarding_sender_query> Tag, class... As>
+                requires stdexec::__callable<Tag, const Sender&, As...>
+            friend auto tag_invoke(Tag tag,
+                                   const bulk_sender& self,
+                                   As&&... as) noexcept(stdexec::__nothrow_callable<Tag, const Sender&, As...>)
+                -> stdexec::__call_result_if_t<stdexec::tag_category<Tag, stdexec::forwarding_sender_query>,
+                                               Tag,
+                                               const Sender&,
+                                               As...> {
+                return ((Tag &&) tag)(self.sndr_, (As &&) as...);
+            }
+        };
+
+        friend sender tag_invoke(stdexec::schedule_t, const scheduler& s) noexcept { return sender{*s.pool_}; }
+
+        template <stdexec::sender Sender, std::integral Shape, class Fun>
+        using bulk_sender_t =
+            bulk_sender<stdexec::__x<std::remove_cvref_t<Sender>>, Shape, stdexec::__x<std::remove_cvref_t<Fun>>>;
+
+        template <stdexec::sender S, std::integral Shape, class Fn>
+        friend bulk_sender_t<S, Shape, Fn> tag_invoke(stdexec::bulk_t,
+                                                      const scheduler& sch,
+                                                      S&& sndr,
+                                                      Shape shape,
+                                                      Fn fun) noexcept {
+            return bulk_sender_t<S, Shape, Fn>{*sch.pool_, (S &&) sndr, shape, (Fn &&) fun};
+        }
+
+        friend stdexec::forward_progress_guarantee tag_invoke(stdexec::get_forward_progress_guarantee_t,
+                                                              const tbb_thread_pool&) noexcept {
+            return stdexec::forward_progress_guarantee::parallel;
```

Comment:
This looks right.

https://spec.oneapi.io/versions/1.0-rev-3/elements/oneTBB/source/task_scheduler/task_arena/task_arena_cls.html

> The task is scheduled for eventual execution by a worker thread even if no thread ever explicitly waits for the task to complete.

Reply:
If it's conservative, which would the less-conservative value be?